consensus/ibft/ibft.go

```diff
@@ -1255,14 +1255,14 @@ func (i *Ibft) VerifyHeader(parent, header *types.Header) error {
 		return err
 	}
 
-	// process the new block in order to update the snapshot
-	if err := i.processHeaders([]*types.Header{header}); err != nil {
-		return err
-	}
-
 	return nil
 }
 
+// ProcessHeaders updates the snapshot based on previously verified headers
+func (i *Ibft) ProcessHeaders(headers []*types.Header) error {
+	return i.processHeaders(headers)
+}
+
```

Comment:
For now, processHeaders is required for only IBFT. Why blockchain needs to call processHeaders? And do you think whether it's possible to call in IBFT? (For example, you call processHeaders after WriteBlock)

Reply:
I agree with Yoshiki 👍 

Is there any way to move out this processing logic to be dependent only on the actual IBFT implementation, and not have the `blockchain` module call it?

Reply:
Initially I tried going in the same direction - moving snapshot updating out of `WriteBlock`. Sadly, there are a few other modules (besides consensus) that are running this method under the assumption it also updates state. (`ProcessHeaders` was in `VerifyHeaders` before).

Making `processHeaders` as part of Ibft only would break this assumption and there also wouldn't be any way to have this logic available at callsite:

restore.go:
[importBlocks](https://github.com/0xPolygon/polygon-edge/blob/86e35dfe6a6e191fd961236784617a4fe0df76d3/archive/restore.go#L76)
syncer.go: [WatchSyncWithPeer](https://github.com/0xPolygon/polygon-edge/blob/86e35dfe6a6e191fd961236784617a4fe0df76d3/protocol/syncer.go#L603)
[BulkWyncWithPeer](https://github.com/0xPolygon/polygon-edge/blob/86e35dfe6a6e191fd961236784617a4fe0df76d3/protocol/syncer.go#L678)

Reply:
@Kourin1996 I think it's best if we leave it here in `WriteBlock`, since it has coupling with `ProcessBlock` - from places where `ProcessBlock` would not be accessible (syncer and restore)